packages/sdk/src/marketplaces/LooksRare.ts

```diff
@@ -0,0 +1,572 @@
+  SupportedChainId.RINKEBY,
+] as const;
+
+const DAY = 60 * 24;
+const DEFAULT_EXPIRATION = 30 * DAY; // Follow LooksRare's default
+
+// Following LooksRare's default
+// Enforce a max slippage of 15% on all orders, if a collection change the fees to be >15%, the order will become invalid
+const DEFAULT_MIN_PERCENTAGE_TO_ASK = 8500;
+
+export class LooksRare implements Marketplace<MakeOrderResult> {
+  private readonly apiKey: string | undefined;
+  private readonly address: string;
+  private readonly chainId: SupportedChainId;
+  private readonly signer: Signer;
+
+  constructor({ apiKey, signer, chainId, address }: _LooksRareConfig) {
+    if (!LooksRare.supportsChainId(chainId)) {
+      throw new Error("unsupported chain id");
+    }
+
+    this.apiKey = apiKey;
+    this.address = address;
+    this.chainId = chainId;
+    this.signer = signer;
+  }
+
+  static supportsChainId(chainId: number): boolean {
+    return SUPPORTED_CHAIN_ID.includes(chainId);
+  }
+
+  /**
+   * Makes an order and post to LooksRare order book.
+   */
+  async makeOrder({
+    makerAssets,
+    takerAssets,
+    taker,
+    expirationTime,
+  }: MakeOrderParams): Promise<MakeOrderResult> {
+    if (taker) {
+      // We disable private sale for now
+      // We will need to include ethers.js lib to support it because LooksRare uses it to verify address in params
+      // (https://github.com/LooksRare/looksrare-sdk/blob/eb2e61dba502f0d35c3c3c20236f6505ef730877/src/sign/encodeOrderParams.ts#L13)
+      // and has listed ethers@5.x as a peer dependency. 0x has ethers as dependency but it is 4.x which is missing
+      // the utils.isAddress method.
+      throw new Error("targeted taker unsupported in looksrare");
+    }
+
+    assertAssetsIsNotEmpty(makerAssets, "maker");
+    assertAssetsIsNotEmpty(takerAssets, "taker");
+    assertAssetsIsNotBundled(makerAssets);
+    assertAssetsIsNotBundled(takerAssets);
+
+    const makerAsset = makerAssets[0];
+    const takerAsset = takerAssets[0];
+    assertAssetsIsNotErc20AndErc20(makerAsset, takerAsset);
+    assertAssetsIsNotErc721Erc1155AndErc721Erc115(makerAsset, takerAsset);
+
+    const now = Math.floor(Date.now() / 1000);
+    const expTime =
+      expirationTime && Math.round(expirationTime.getTime() / 1000);
+    const addresses = addressesByNetwork[this.chainId];
+    const signerAddress = this.address;
+
+    let baseAsset: Erc721Asset | Erc1155Asset;
+    let quoteAsset: Erc20Asset;
+    let isOrderAsk: boolean; // true = Ask, false = Bid
+
+    if (
+      (makerAsset.type === "ERC721" || makerAsset.type === "ERC1155") &&
+      takerAsset.type === "ERC20"
+    ) {
+      baseAsset = makerAsset;
+      quoteAsset = takerAsset;
+      isOrderAsk = true;
+    } else if (
+      makerAsset.type === "ERC20" &&
+      (takerAsset.type === "ERC721" || takerAsset.type === "ERC1155")
+    ) {
+      baseAsset = takerAsset;
+      quoteAsset = makerAsset;
+      isOrderAsk = false;
+    } else {
```

Comment:
Can this final else actually happen here, since we have assets above, that suppose to protect us from everything else? Is it possible to rewrite it like this:
```
const isOrderAsk = takerAsset.type === "ERC20";
const baseAsset = isOrderAsk ? makerAsset : takerAsset;
const quoteAsset = isOrderAsk ? takerAsset : makerAsset;
```

Or am I missing some edge-case?

Reply:
The whole chunk really is for typescript to discriminate which one is erc 721/1155 and which one is erc 20, otherwise typescript would have a hard time telling between the two and just assume both are "Asset"s
These lines (and more) would then fail type checking:
https://github.com/gomuweb3/gomu/pull/9/files/9a1ed1936928cc0c67fe2c17f78827ff5872821f#diff-867ceb7affd6e28b753e53127a0c5a44a39003b1c3506a499927588bd94712e5R280-R281
because typescript isn't sure it's an erc20 asset and should therefore have those 2 properties

I get your concern tho and agree it's a hefty chunk, I suppose we can refactor the chunk into a common util function that spits out base asset and quote asset

Reply:
Anw I ask that we put the refactoring concern aside first and stick with this in the interest of time if there're no breaking bugs.

What we can later do in the upcoming lull period (during NFT NYC week?) is to write up unit tests and then refactor. The refactor will extend to opensea marketplace sdk since there are similar chunks inside.